pandas-stubs/core/series.pyi

```diff
@@ -1904,6 +1894,15 @@ class TimestampSeries(Series[Timestamp]):
     def dt(self) -> TimestampProperties: ...  # type: ignore[override]
     def __add__(self, other: TimedeltaSeries | np.timedelta64) -> TimestampSeries: ...  # type: ignore[override]
     def __radd__(self, other: TimedeltaSeries | np.timedelta64) -> TimestampSeries: ...  # type: ignore[override]
+    @overload  # type: ignore[override]
+    def __sub__(
+        self, other: Timestamp | datetime | TimestampSeries
+    ) -> TimedeltaSeries: ...
+    @overload
+    def __sub__(
+        self,
+        other: Timedelta | TimedeltaSeries | TimedeltaIndex | np.timedelta64,
```

Comment:
Intuitively I would expect that `Timedelta - Timedelta` is still a `Timedelta`? Might be good to have a test for this as this overload changes this behavior.

Reply:
Oh, sorry I didn't realize this overload is in TimestampSeries